TST: Convolution and coordinates failures with numpy dev

Example log: https://travis-ci.org/astropy/astropy/jobs/350000086

```
______ TestConvolve2D.test_unity_3x3_withnan[wrap-interpolate-True-False] ______
self = <astropy.convolution.tests.test_convolve_fft.TestConvolve2D object at 0x7f7f08a3e630>
boundary = 'wrap', nan_treatment = 'interpolate', normalize_kernel = True
preserve_nan = False
    @pytest.mark.parametrize(option_names_preserve_nan, options_preserve_nan)
    def test_unity_3x3_withnan(self, boundary, nan_treatment,
                               normalize_kernel, preserve_nan):
...
>           raise ValueError("Unsupported dtype %s" % x.dtype)
E           ValueError: Unsupported dtype float16
/home/travis/miniconda/envs/test/lib/python3.6/site-packages/numpy/testing/nose_tools/utils.py:1585: ValueError
```

```
_____ TestConvolve2D.test_unity_3x3_withnan[wrap-interpolate-False-False] ______
self = <astropy.convolution.tests.test_convolve_fft.TestConvolve2D object at 0x7f7f5dbc5f60>
boundary = 'wrap', nan_treatment = 'interpolate', normalize_kernel = False
preserve_nan = False
    @pytest.mark.parametrize(option_names_preserve_nan, options_preserve_nan)
    def test_unity_3x3_withnan(self, boundary, nan_treatment,
                               normalize_kernel, preserve_nan):
...
>           raise ValueError("Unsupported dtype %s" % x.dtype)
E           ValueError: Unsupported dtype float16
/home/travis/miniconda/envs/test/lib/python3.6/site-packages/numpy/testing/nose_tools/utils.py:1585: ValueError
```

```
_____________________ [doctest] docs/coordinates/index.rst _____________________
248 :func:`astropy.coordinates.EarthLocation.get_site_names`.
249 
250 For arbitrary Earth addresses (e.g., not observatory sites), use the
251 `~astropy.coordinates.EarthLocation.of_address` classmethod. Any address passed
252 to this function uses Google maps to retrieve the latitude and longitude and can
253 also (optionally) query Google maps to get the height of the location. As with
254 Google maps, this works with fully specified addresses, location names, city
255 names, and etc.::
256 
257     >>> EarthLocation.of_address('1002 Holy Grail Court, St. Louis, MO')  # doctest: +REMOTE_DATA +FLOAT_CMP
Expected:
    <EarthLocation (-26726.98216371, -4997009.8604809, 3950271.16507911) m>
Got:
    <EarthLocation (-26724.93406367, -4997006.87829221, 3950274.92599205) m>
/tmp/astropy-test-993gsz3o/lib/python3.6/site-packages/docs/coordinates/index.rst:257: DocTestFailure
```
c/c @mhvk

--- a/astropy/coordinates/tests/test_earth.py
+++ b/astropy/coordinates/tests/test_earth.py
@@ -4,7 +4,6 @@
 """Test initialization of angles not already covered by the API tests"""
 
 import pickle
-import time as _time
 
 import pytest
 import numpy as np
@@ -299,10 +298,17 @@ def test_of_address():
         EarthLocation.of_address("lkjasdflkja")
 
     # a location and height
-    loc = EarthLocation.of_address("New York, NY", get_height=True)
-    assert quantity_allclose(loc.lat, 40.7128*u.degree)
-    assert quantity_allclose(loc.lon, -74.0059*u.degree)
-    assert quantity_allclose(loc.height, 10.438659669*u.meter, atol=1.*u.cm)
+    try:
+        loc = EarthLocation.of_address("New York, NY", get_height=True)
+    except NameResolveError as e:
+        # Buffer above insufficient to get around API limit but
+        # we also do not want to drag things out with time.sleep(0.195),
+        # where 0.195 was empirically determined on some physical machine.
+        pytest.xfail(str(e))
+    else:
+        assert quantity_allclose(loc.lat, 40.7128*u.degree)
+        assert quantity_allclose(loc.lon, -74.0059*u.degree)
+        assert quantity_allclose(loc.height, 10.438659669*u.meter, atol=1.*u.cm)
 
 
 def test_geodetic_tuple():

--- a/astropy/coordinates/tests/test_earth.py
+++ b/astropy/coordinates/tests/test_earth.py
@@ -1,11 +1,10 @@
 # -*- coding: utf-8 -*-
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
-
-
 """Test initialization of angles not already covered by the API tests"""
 
 import pickle
+import time as _time
 
 import pytest
 import numpy as np
@@ -288,15 +287,16 @@ def test_repr_latex():
 
 @pytest.mark.remote_data
 def test_of_address():
-    # no match
-    with pytest.raises(NameResolveError):
-        EarthLocation.of_address("lkjasdflkja")
-
     # just a location
     loc = EarthLocation.of_address("New York, NY")
     assert quantity_allclose(loc.lat, 40.7128*u.degree)
     assert quantity_allclose(loc.lon, -74.0059*u.degree)
     assert np.allclose(loc.height.value, 0.)
+
+    # Put this one here as buffer to get around Google map API limit per sec.
+    # no match
+    with pytest.raises(NameResolveError):
+        EarthLocation.of_address("lkjasdflkja")
 
     # a location and height
     loc = EarthLocation.of_address("New York, NY", get_height=True)

--- a/astropy/coordinates/tests/test_earth.py
+++ b/astropy/coordinates/tests/test_earth.py
@@ -298,7 +298,7 @@ def test_of_address():
         assert np.allclose(loc.height.value, 0.)
 
     # Put this one here as buffer to get around Google map API limit per sec.
-    # no match
+    # no match: This always raises NameResolveError
     with pytest.raises(NameResolveError):
         EarthLocation.of_address("lkjasdflkja")
 

--- a/astropy/coordinates/tests/test_earth.py
+++ b/astropy/coordinates/tests/test_earth.py
@@ -287,10 +287,15 @@ def test_repr_latex():
 @pytest.mark.remote_data
 def test_of_address():
     # just a location
-    loc = EarthLocation.of_address("New York, NY")
-    assert quantity_allclose(loc.lat, 40.7128*u.degree)
-    assert quantity_allclose(loc.lon, -74.0059*u.degree)
-    assert np.allclose(loc.height.value, 0.)
+    try:
+        loc = EarthLocation.of_address("New York, NY")
+    except NameResolveError as e:
+        # Google map API limit might surface even here in Travis CI.
+        pytest.xfail(str(e))
+    else:
+        assert quantity_allclose(loc.lat, 40.7128*u.degree)
+        assert quantity_allclose(loc.lon, -74.0059*u.degree)
+        assert np.allclose(loc.height.value, 0.)
 
     # Put this one here as buffer to get around Google map API limit per sec.
     # no match
@@ -301,7 +306,7 @@ def test_of_address():
     try:
         loc = EarthLocation.of_address("New York, NY", get_height=True)
     except NameResolveError as e:
-        # Buffer above insufficient to get around API limit but
+        # Buffer above sometimes insufficient to get around API limit but
         # we also do not want to drag things out with time.sleep(0.195),
         # where 0.195 was empirically determined on some physical machine.
         pytest.xfail(str(e))

--- a/docs/coordinates/index.rst
+++ b/docs/coordinates/index.rst
@@ -254,12 +254,14 @@ also (optionally) query Google maps to get the height of the location. As with
 Google maps, this works with fully specified addresses, location names, city
 names, and etc.::
 
-    >>> EarthLocation.of_address('1002 Holy Grail Court, St. Louis, MO')  # doctest: +REMOTE_DATA +FLOAT_CMP
+.. doctest-skip::
+
+    >>> EarthLocation.of_address('1002 Holy Grail Court, St. Louis, MO')
     <EarthLocation (-26726.98216371, -4997009.8604809, 3950271.16507911) m>
     >>> EarthLocation.of_address('1002 Holy Grail Court, St. Louis, MO',
-    ...                          get_height=True)  # doctest: +REMOTE_DATA +FLOAT_CMP
+    ...                          get_height=True)
     <EarthLocation (-26727.6272786, -4997130.47437768, 3950367.15622108) m>
-    >>> EarthLocation.of_address('Danbury, CT')  # doctest: +REMOTE_DATA +FLOAT_CMP
+    >>> EarthLocation.of_address('Danbury, CT')
     <EarthLocation ( 1364606.64511651, -4593292.9428273,  4195415.93695139) m>
 
 .. note::

